src/services/webapi/webapi.service.ts

```diff
@@ -94,10 +103,14 @@ export class WizdomWebApiService implements IWizdomWebApiService {
             else
             {
                 console.error("Corsproxy request ratelimit exceeded. More than " + requestRateLimitCount + " was made over a period of " + (requestRateLimitTimeout/1000) + " seconds");
-                fail("Corsproxy request ratelimit exceeded");
+                fail({errorType: WebApiErrorType.RateLimitExeeded, message: "Corsproxy request ratelimit exceeded"});
```

Comment:
This change failes the test "should ratelimit request, if to more than 300 GET requests is made in 5 min" in webapi.service.spec.ts.
The test needs to be updated to the new fail object.